src/gt4py/next/program_processors/runners/dace_fieldview/transformations/k_blocking.py

```diff
@@ -0,0 +1,439 @@
+
+import copy
+import functools
+from typing import Any, Optional, Union
+
+import dace
+from dace import properties, subsets, transformation
+from dace.sdfg import SDFG, SDFGState, graph as dace_graph, nodes
+from dace.transformation import helpers
+
+from gt4py.next import common as gtx_common
+from gt4py.next.program_processors.runners.dace_fieldview import utility as dace_fieldview_util
+
+
+@properties.make_properties
+class KBlocking(transformation.SingleStateTransformation):
+    """Applies k-Blocking with separation on a Map.
+
+    This transformation takes a multidimensional Map and performs blocking on a
+    dimension, that is commonly called "k", but identified with `block_dim`.
+
+    All dimensions except `k` are unaffected by this transformation. In the outer
+    Map the will replace the `k` range, currently `k = 0:N`, with
+    `__coarse_k = 0:N:B`, where `N` is the original size of the range and `B`
+    is the block size, passed as `blocking_size`. The transformation also handles the
+    case if `N % B != 0`.
+    The transformation will then create an inner sequential map with
+    `k = __coarse_k:(__coarse_k + B)`.
+
+    However, before the split the transformation examines all adjacent nodes of
+    the original Map. If a node does not depend on `k`, then the  node will be
+    put between the two maps, thus its content will only be computed once.
+
+    The function will also change the name of the outer map, it will append
+    `_blocked` to it.
+    """
+
+    blocking_size = properties.Property(
+        dtype=int,
+        allow_none=True,
+        desc="Size of the inner k Block.",
+    )
+    block_dim = properties.Property(
+        dtype=str,
+        allow_none=True,
+        desc="Which dimension should be blocked (must be an exact match).",
+    )
+
+    map_entry = transformation.transformation.PatternNode(nodes.MapEntry)
+
+    def __init__(
+        self,
+        blocking_size: Optional[int] = None,
+        block_dim: Optional[Union[gtx_common.Dimension, str]] = None,
+    ) -> None:
+        super().__init__()
+        if isinstance(block_dim, str):
+            pass
+        elif isinstance(block_dim, gtx_common.Dimension):
+            block_dim = dace_fieldview_util.get_map_variable(block_dim)
+        if block_dim is not None:
+            self.block_dim = block_dim
+        if blocking_size is not None:
+            self.blocking_size = blocking_size
+
+    @classmethod
+    def expressions(cls) -> Any:
+        return [dace.sdfg.utils.node_path_graph(cls.map_entry)]
+
+    def can_be_applied(
+        self,
+        graph: Union[SDFGState, SDFG],
+        expr_index: int,
+        sdfg: dace.SDFG,
+        permissive: bool = False,
+    ) -> bool:
+        """Test if the map can be blocked.
+
+        The test involves:
+        - Toplevel map.
+        - The map shall not be serial.
+        - The block dimension must be present (exact match).
+        - The map range must have stride one.
```

Comment:
How do you check this requirement? The map itself does not have a stride. From what I see below you require that the range step is 1.

Reply:
The map has a stride, a range object is essentially a list of length 3 tuple:
- initial value
- end value (including, so it is not pass the end).
- step size (stride).

Reply:
Yes, I was just confused by the name (I usually call it step).

Reply:
Steps sounds actually better.